Update LLVM (2025-10-08)

rniwa opened the pull request:
#### 48d99a3d98c73c1668a3fc0afe684ee19e7634e9
<pre>
Update LLVM (2025-10-08)
<a href="https://bugs.webkit.org/show_bug.cgi?id=300379">https://bugs.webkit.org/show_bug.cgi?id=300379</a>

Reviewed by Brianna Fan and Chris Dumez.

Update the clang static analyzer for WebKit. This update includes fixes to
NoUnretainedMemberChecker so that @property is only checked when @implementation exists,
allow `leakRef()` in a Objective-C create/copy functions which returns +1, and treat
`NSApp` and a boxed value such as @(foo) as a constant.

* Tools/CISupport/safer-cpp-llvm-version:

Canonical link: <a href="https://commits.webkit.org/301221@main">https://commits.webkit.org/301221@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ac60a48b0957ade1515a5662b9cd853774be2f8c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44892 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35631 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132076 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77086 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/897ea341-270c-42f2-9622-83bb92655301) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127102 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45583 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53453 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95335 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63275 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b6e96e16-eabc-4611-a350-32a5bdf7129f) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128179 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36390 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111985 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75874 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bc6a46ff-d0d9-43cf-b7f0-39b851a5e931) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35286 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30153 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75554 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106157 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30378 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134757 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52034 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39812 "Found 5 new test failures: http/tests/media/modern-media-controls/overflow-support/playback-speed-live-broadcast.html http/tests/media/reload-after-dialog.html http/tests/media/track/track-webvtt-slow-loading-2.html imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/utf-8.html?include=loading imported/w3c/web-platform-tests/html/infrastructure/urls/resolving-urls/query-encoding/windows-1252.html?include=loading (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103802 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52469 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108206 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103571 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48922 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27218 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/49120 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19618 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51926 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51288 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54644 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52982 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ac60a48b0957ade1515a5662b9cd853774be2f8c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51999

rniwa committed:
Update LLVM (2025-10-08)
https://bugs.webkit.org/show_bug.cgi?id=300379

Reviewed by Brianna Fan and Chris Dumez.

Update the clang static analyzer for WebKit. This update includes fixes to
NoUnretainedMemberChecker so that @property is only checked when @implementation exists,
allow `leakRef()` in a Objective-C create/copy functions which returns +1, and treat
`NSApp` and a boxed value such as @(foo) as a constant.

* Tools/CISupport/safer-cpp-llvm-version:

Canonical link: https://commits.webkit.org/301221@main
